urdfModifiers/core/fixedOffsetModifier.py

```diff
@@ -0,0 +1,147 @@
+        """Returns the significant length, for spheres it returns diameter instead of radius"""
+        significant_length = self.link_modifier.get_significant_length()
+        geometry_type, _ = self.get_geometry(self.link_modifier.get_visual())
+        if geometry_type == Geometry.SPHERE:
+            significant_length *= 2
+        return significant_length
+
+    def get_joint_origin(self, joint):
+        """Returns the origin of a joint"""
+        return (matrix_to_xyz_rpy(joint.origin) if joint else None)
+
+    def get_link_origin(self, link):
+        """Returns the origin of a joint"""
+        return matrix_to_xyz_rpy(link.visuals[0].origin)
+
+    def calculate_offsets(self):
+        link_length = self.get_significant_length()
+        link_origin = self.get_link_origin(self.link)
+        link_origin_z = link_origin[2]
+        child_joint_origin = self.get_joint_origin(self.child_joint)
+        child_joint_origin_z = (child_joint_origin[2] if child_joint_origin is not None else 0)
+        parent_joint_offset = (link_origin_z - link_length * math.copysign(1, child_joint_origin_z) / 2 if self.parent_joint else None)
+        child_joint_offset = (link_origin_z + link_length * math.copysign(1, child_joint_origin_z) / 2 - child_joint_origin_z if self.child_joint else None)
+        return parent_joint_offset, child_joint_offset
+
+    def modify(self, modifications):
+        """Performs the modifications in the link-joint setup"""
+
+        trivial_modifications = Modification()
+        if modifications.radius:
+            trivial_modifications.add_radius(modifications.radius.value, modifications.radius.absolute)
+        if modifications.density:
+            trivial_modifications.add_density(modifications.density.value, modifications.density.absolute)
+        if modifications.mass:
+            trivial_modifications.add_mass(modifications.mass.value, modifications.mass.absolute)
+        
+        self.link_modifier.modify(trivial_modifications)
+
+        if modifications.dimension:
+            original_length = self.get_significant_length()
+            if modifications.dimension.absolute:
+                self.change_dimension_and_keep_offsets(modifications.dimension.value)
+            else:
+                self.change_dimension_and_keep_offsets(original_length * modifications.dimension.value)
+
+    def change_dimension_and_keep_offsets(self, new_length):
+        """Changes the dimension of the link while keeping the offset between it and both parent and child joints"""
+
+        child_joint_origin = self.get_joint_origin(self.child_joint)
+        child_joint_origin_z = (child_joint_origin[2] if child_joint_origin is not None else 0)
+
+        parent_joint_offset, child_joint_offset = self.calculate_offsets()
+
+        link_modification = Modification()
+
+        if parent_joint_offset is not None:
+            new_link_origin = parent_joint_offset + new_length * math.copysign(1, child_joint_origin_z) / 2
```

Comment:
I would add a reference to this formula. It is not trivial to derive it (at least, not for me and possible future people that may modify this class) so I think it is useful to add some reference (even just a comment to the issue in which the formula where derived). Code is tipically read many more times then it is edited.